src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1208,19 +1223,23 @@ private static ExecutorService getExecutorService() {
     }
 
     private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy() {
-        return new PeriodicShardSyncStrategy(createPeriodicShardSyncManager());
+        return new PeriodicShardSyncStrategy(leaderElectedPeriodicShardSyncManager);
     }
 
-    private ShardEndShardSyncStrategy createShardEndShardSyncStrategy(ShardSyncTaskManager shardSyncTaskManager) {
-        return new ShardEndShardSyncStrategy(shardSyncTaskManager);
+    private ShardEndShardSyncStrategy createShardEndShardSyncStrategy() {
+        return new ShardEndShardSyncStrategy(controlServer, leaderElectedPeriodicShardSyncManager);
```

Comment:
Let's rename this as shardEndShardSyncManager. controlServer is very confusing.

Reply:
Agreed, will rename.